src/Microsoft.Azure.WebJobs.Host/JobHostConfiguration.cs

```diff
@@ -123,6 +125,31 @@ public string HostId
             }
         }
 
+        /// <summary>
+        /// Gets the host machine ID. All host instances on the same machine will have the same HostMachineId.
+        /// </summary>
+        public string HostMachineId
+        {
+            get
+            {
+                if (_hostMachineId == null)
+                {
+                    _hostMachineId = Environment.GetEnvironmentVariable("WEBSITE_INSTANCE_ID")
+                        ?? Environment.MachineName.GetHashCode().ToString("X", CultureInfo.InvariantCulture).PadLeft(32, '0');
+
+                    _hostMachineId = _hostMachineId.Substring(0, 32).ToLowerInvariant();
+                }
+
+                return _hostMachineId;
+            }
+
+            // for testing
+            internal set
```

Comment:
If we make this public, we'll also probably want to put validation in place similar to what we have for host id. Since this value will be used in queue names and storage entities, only valid storage characters are allowed.